src/components/Dialogs/ConvertTmDialog.jsx

```diff
@@ -88,7 +88,65 @@ const createHtmlInputs = (td) => {
                 </div>
             );
         });
-        return htmlInputs;
+
+        // Containers for created html elements (checkboxes) of each interaction affordance
+        let htmlProperties = [];
+        let htmlActions = [];
+        let htmlEvents = [];
+
+        try {
+            const parsed = JSON.parse(td);
+            const properties = Object.keys(parsed["properties"] ? parsed["properties"] : {});
+            const actions = Object.keys(parsed["actions"] ? parsed["actions"] : {});
+            const events = Object.keys(parsed["events"] ? parsed["events"] : {});
+            const requiredFields = {"properties": [], "actions": [], "events": []};
+
+            // Parse the required interaction affordances
+            if (parsed["tm:required"]) {
+                for (const field of parsed["tm:required"]) {
+                    if (field.startsWith("#properties/"))
+                        requiredFields["properties"].push(field.split("/")[1]);
+                    else if (field.startsWith("#actions/"))
+                        requiredFields["actions"].push(field.split("/")[1]);
+                    else if (field.startsWith("#events/"))
+                        requiredFields["events"].push(field.split("/")[1]);
+                }
+            }
+
+            // Create html (checkboxes) for specific interaction affordances
+            function createAffordanceHtml(affName, affContainer) {
+                return affContainer.map((aff) => {
+                    const required = requiredFields[affName].includes(aff);
+                    return (
+                        <div key={`${affName}/${aff}`} className="form-checkbox py-1 pl-2">
+                            <input id={`${affName}/${aff}`}
+                                className="form-checkbox-input"
+                                type="checkbox"
+                                value={`#${affName}/${aff}`}
+                                disabled={required}
+                                defaultChecked={true}
+                                title={required ? "This field is required by the TM." : ""}
+                                data-interaction={affName}
+                            />
+                            <label className="form-checkbox-label pl-2" htmlFor={`${affName}/${aff}`}>{`#${affName}/${aff}`}</label>
+                        </div>
+                    );
+                });
+            };
+
+            htmlProperties = createAffordanceHtml("properties", properties);
+            htmlActions = createAffordanceHtml("actions", actions);
+            htmlEvents = createAffordanceHtml("events", events);
+
+        } catch (ignored) {}
```

Comment:
Would be nice to at least give the user a hint that his TM could not be parsed.

Reply:
If something is wrong with parsing they are already getting this banner, it's triggered by the editor automatically.
![image](https://user-images.githubusercontent.com/14371560/123878258-0b426600-d93f-11eb-8abe-3b51a87d1a23.png)

Reply:
OK